### PR TITLE
Challenge5

### PR DESCRIPTION
**Pull request by kzivanovic**

### CODE REVIEW

File: cypress/support/commands.js [hunk 1]
```diff
@@ -22,4 +22,13 @@
...
\ No newline at end of file
+// Cypress.Commands.overwrite('visit', (originalFn, url, options) => { ... })
+
+
+Cypress.Commands.add('login',(username, password)=>{
+    cy.visit('https://www.saucedemo.com/')
+    cy.get('[data-test="username"]').type(username)
+    cy.get('[data-test="password"]').type(password)
+    cy.get('[data-test="login-button"]').click()
+    cy.url().should('contains', 'inventory.html')
```
Comment (stefanworkco) [inline cypress/support/commands.js]: remove this check as assertion is not done in custom commands but in spec files

File: cypress/e2e/cart.cy.js [hunk 2]
```diff
@@ -0,0 +1,66 @@
+describe('Error with no data',()=>{
```
Comment (stefanworkco) [inline cypress/e2e/cart.cy.js]: Refactor code to have just 1 describe with 4 it() (also use beforeEach in that case)

File: cypress/e2e/cart.cy.js [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+        cy.get('.shopping_cart_link').click()
+        cy.get('[data-test="checkout"]').click()
+        cy.get('[data-test="continue"]').click()
+        cy.get('.error-message-container').should('have.text','Error: First Name is required');
+    })
+})
+
+describe('Successful shopping flow',()=>{
+    it('Successful shopping flow',()=>{
+        cy.login('standard_user','secret_sauce')
```
Comment (stefanworkco) [inline cypress/e2e/cart.cy.js]: when you add beforeEach, you don't need this line ( `cy.login('standard_user','secret_sauce')` ) here

**Review by stefanworkco — CHANGES_REQUESTED** [review]

File: cypress/e2e/cart.cy.js [hunk 4]
```diff
@@ -0,0 +1,62 @@
+describe('Swag Labs tests',()=>{
+  beforeEach(()=>{
+    cy.login('standard_user','secret_sauce')
+  })
+    it('Error wih no data',()=>{
+        cy.login('standard_user','secret_sauce')
```
Comment (stefanworkco) [inline cypress/e2e/cart.cy.js]: remove this since you already do this step in beforeEach (same in second it)

File: cypress/e2e/cart.cy.js [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+
+it('Removing items from bag',()=>{
+     it('Removing items from bag',()=>{
+        cy.get('[data-test="add-to-cart-sauce-labs-backpack"]').click()
+        cy.get('.shopping_cart_badge').should('be.visible')
+        cy.get('.shopping_cart_link').click()
+        cy.get('[data-test="remove-sauce-labs-backpack"]').click()
+        cy.get('.shopping_cart_badge').should('not.exist')
+      })
+  it('Continue shopping',()=>{
```
Comment (stefanworkco) [inline cypress/e2e/cart.cy.js]: extra it() wrapper

**Review by stefanworkco — APPROVED** [review]

**Review by stefanworkco — COMMENTED** [review]
Changes required

**Review by stefanworkco — APPROVED** [review]

**Review by vesnaatwork — APPROVED** [review]
👏

**Review by tamaracvorovic-work — APPROVED** [review]
👏 🎉

**Pull request merged**